fmn/consumer/backends/sse.py

```diff
@@ -0,0 +1,171 @@
+    might arrive: the `raw` message from fedmsg itself and the product of a
+    call to `fedmsg.meta.conglomerate(..)`
+
+    example output from sse.py is below
+
+      {
+        "dom_id": "d38b2b6c-a3c9-4772-b6aa-0a70a6bee517",
+        "date_time": "2008-09-03T20:56:35.450686Z",
+        "icon": "https://apps.fedoraproject.org/packages/images/icons/package_128x128.png",
+        "link": "https://pagure.io/<repo>/issue/148703615",
+        "markup": "<a href="pagure.io/atelic">@atelic</a> opened a new ticket #148703615: \"Things are broken wat do\"",
+        "secondary_icon": null,
+      }
+    """
+
+    dom_id = six.text_type(uuid.uuid4())
+    date_time = ''
+    icon = ''
+    link = ''
+    markup = ''
+    secondary_icon = ''
+    username = ''
+    subtitle = ''
+    if not 'subtitle' in msg:
+        # This handles normal, 'raw' messages which get passed through msg2*.
+        # Tack a human-readable delta on the end so users know that fmn is
+        # backlogged (if it is).
+        if msg['timestamp']:
+            date_time = msg['timestamp']
+            date_time = datetime.datetime.fromtimestamp(date_time)
+            date_time = date_time.isoformat()
+        else:
+            date_time = datetime.datetime.utcnow().isoformat()
+
+        icon = fedmsg.meta.msg2icon(msg, **config)
+        link = fedmsg.meta.msg2link(msg, **config)
+        secondary_icon = fedmsg.meta.msg2secondary_icon(msg, **config)
+        username = fedmsg.meta.msg2agent(msg, **config)
+        title = fedmsg.meta.msg2title(msg, **config)
+        subtitle = fedmsg.meta.msg2subtitle(msg, **config)
+
+    else:
+        # This handles messages that have already been 'conglomerated'.
+        title = u""
+        subtitle = msg['subtitle']
+        link = msg['link']
+
+    if recipient['shorten_links']:
+        link = shorten(link)
+    event_link = ''
+    if title and link:
+        event_link = '<a href="' + link + '">' + title + '</a>'
+    user_link = ''
+    if username:
+        user_link = '<a href="/'+str(username).replace("@", "")+'">'+username+'</a>'
+
+    markup = user_link + ' ' + event_link + ' ' + subtitle
+    output = {"dom_id": str(dom_id),
+              "date_time": str(date_time),
+              "icon": str(icon),
+              "link": str(link),
+              "markup": str(markup),
+              "secondary_icon": str(secondary_icon)
+              }
+
+    return json.dumps(output)
+
+
+class SSEBackend(BaseBackend):
+    __context_name__ = "sse"
+
+    def __init__(self, *args, **kwargs):
+        super(SSEBackend, self).__init__(*args, **kwargs)
+
+    def send_msg(self, msg="hello world", username="user.id.fedoraproject.org"):
+        username = str(username)
+        self.log.debug("SSE: \t Sending msg to pika to queue " + username)
+        ms_24hr = 24 * 60 * 60 *1000
+        CONFIG = fedmsg.config.load_config()
+        host = CONFIG.get('fmn.sse.pika.host', 'localhost')
+        exchange = 'user'
```

Comment:
for now I am forcing it to send to exchange 'user' as fedmsg for 'groups' is not yet set up in fedmsg